src/main/java/org/jabref/logic/pdf/search/indexing/DocumentReader.java

```diff
@@ -116,24 +125,25 @@ private boolean isValidField(String value) {
         return !(StringUtil.isNullOrEmpty(value));
     }
 
-    private void addContentIfNotEmpty(PDDocument pdfDocument, Document newDocument) {
+    private static String mergeLines(String text) {
+        return text.replaceAll("\\-\n", "").replaceAll("([^\\.])\n", "$1 ");
```

Comment:
extract to a Pattern and call Pattern.compile()...

regex_PATTERN.matcher(str).replaceAll(repl)